handler/artifact_upload_handler.py

```diff
@@ -45,24 +49,27 @@ def _check_token(self, token):
         # Reduce the risk of timing analysis attacks.
         return hmac.compare_digest(self.hash_expected, hash_result)
 
-    def _index_channel(self):
+    def _conda_index(self, directory):
         """
-        Index the conda channel at `self.write_path` to ensure the channel
-        remains consistent.
+        Update the package index metadata files in the provided directory.
 
         """
-        cmds = [self.conda_exe, 'index', self.write_path]
+        cmds = [self.conda_exe, 'index', directory]
         subprocess.check_call(cmds)
 
     def post(self):
         token = self.get_argument('token')
+        arch = self.get_argument('arch', default='none')
```

Comment:
We should be using some much more defensive code here. I think you may have introduced a security vulnerability...

```
POST: arch='../../passwd'
```

Would be completely valid (and unchecked) input by the looks of it.